### PR TITLE
Added API rest section.

### PR DESCRIPTION
**Pull request by markodoseninfinum**

### CODE REVIEW

File: Api/REST.md [hunk 1]
```diff
@@ -1,2 +1,218 @@
...
 
+We follow RESTFUL conventions in order to have a consistency between client and server communication. We are always using JSON as standard way of communicating for making the request and getting the response.
+
+Here are some guidelines for the best practices.
+
+
+
+### Routes
+
+Standard route format, for and API that returns costumers information, should be *`api/v1/customers`*.  We achieve this by using *`Route`* attribute and .NET routing keywords such as *`{{version:apiVersion}}`*, *`[controller]`* etc.
```
Comment (rubber-duck) [inline Api/REST.md:11]: `for an API ?`

Comment (rubber-duck) [inline Api/REST.md:11]: `customers`

File: Api/REST.md [hunk 2]
```diff
@@ -1,2 +1,218 @@
...
+
+Default route should be set ether on controller level or on base controller if controllers are inheriting it.
```
Comment (rubber-duck) [inline Api/REST.md:13]: Maybe give an example, eg. 
Default route can be set on controller level :
```c#
    [ApiController]
    [Route("api/v{version:apiVersion}/customers")]
    public class CustomersController : ControllerBase
    {
...
    }
```

or specified generically on through a controller base class : ...

Mislim da mozes i taj BaseApiController stavit abstract ?

File: Api/REST.md [hunk 3]
```diff
@@ -1,2 +1,218 @@
...
+
+
+
+### Methods
+
+In .NET there are attributes for each RESTFUL method. Most common that are used are *`GET`*, *`POST`*, *`PUT`* and *`DELETE`*. Methods without a route will take a default route from the controller, but we can also add a route to the method it self.
+
+#### GET
+
+Get is a method reserved for getting information about a resource.
```
Comment (rubber-duck) [inline Api/REST.md:56]: Maybe worth mentioning that GET should not have any side effects (it can be called any number of times and data shouldn't change because of it).

File: Api/REST.md [hunk 4]
```diff
@@ -1,2 +1,218 @@
...
+       return Ok(customer);
+    }
+```
+
+#### Put
+
+Put is a method reserved for updating a resource.
+
+```c#
+	[HttpPut]
```
Comment (rubber-duck) [inline Api/REST.md:85]: Put usually accepts "{customerId}" just like Delete example ?

File: Api/REST.md [hunk 5]
```diff
@@ -1,2 +1,218 @@
...
+Get is a method reserved for getting information about a resource.
+
+```c#
+	[HttpGet]
+    public async Task<IActionResult> Get()
+    {
+       var customers = await _customerServices.GetAllCustomers();
+       return Ok(customers);
+    }
+```
```
Comment (rubber-duck) [inline Api/REST.md:65]: Can also add single customer get example

File: Api/REST.md [hunk 6]
```diff
@@ -1,2 +1,218 @@
...
+
+```c#
+	[HttpPut]
+    public async Task<IActionResult> Put(Customer customer)
+    {
+       var customer = await _customerServices.UpdateCustomer(customer);
+       return Ok(customer);
+    }
+```
+
```
Comment (rubber-duck) [inline Api/REST.md:92]: Maybe mention PATCH for partial updates after PUT ?

File: Api/REST.md [hunk 7]
```diff
@@ -1,2 +1,218 @@
...
+This endpoint now serves two routes: *`api/v1/customers/{customerId}`*, *`api/v1/customers/{customerId}?includeOrders={value}`* where *`{cusomerId}`* is GUID and *`value`* is Boolean (true/false) value.
+
+
+
+### Custom routes
+
+A custom route for action can be defined on action level. The route will inherit controllers route and add its route as a sufix.
```
Comment (rubber-duck) [inline Api/REST.md:185]: "The route will inherit controllers route and add its route as a sufix." I reword this to : if the route is relative (doesn't start with `/` or `~/`) it will be relative to Controller route - and then give some examples of relative and absolute routes

**Comment by rubber-duck** [comment]
@markodoseninfinum @vperko I think it would be good to link https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md here - I think those rest guidelines are quite reasonable

**Comment by markodoseninfinum** [comment]
Resolved all suggestions from Rafael. Also added a documentation that links to Microsoft RESTFUL guidelines.

**Review by rubber-duck — APPROVED** [review]

**Review by vperko — APPROVED** [review]
